ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/StaticQueryExecutor.java

```diff
@@ -119,6 +119,16 @@ public static void validate(
   ) {
     final Query queryStmt = statement.getStatement();
 
+    if (!statement.getConfig().getBoolean(KsqlConfig.KSQL_PULL_QUERIES_ENABLE_CONFIG)) {
+      throw new KsqlRestException(
+          Errors.badStatement(
+              "Pull queries are disabled on this KSQL server - please set "
+                  + KsqlConfig.KSQL_PULL_QUERIES_ENABLE_CONFIG + "=true to enable this feature. "
+                  + "If you intended to issue a push query, resubmit the query with the "
```

Comment:
It's not just a matter of resubmitting the query with `EMIT CHANGES`, right? I thought push queries went to the `/query` endpoint, whereas pull queries went to the `/ksql` endpoint.

Reply:
yes and no - the CLI will handle that for you. if you submit the query to the wrong endpoint, it will also complain. since this is changing with #3742, I'd rather not bake too much of it into the error message

Reply:
Fair enough. As long as our error messages if pull/push queries are submitted to the wrong endpoint are good enough for non-CLI users to know how to respond, that works for me.